src/devices/lightbulbs.ts

```diff
@@ -0,0 +1,546 @@
+        payloadVersion: 1,
+      },
+    };
+
+    // Make request
+    const push = await this.platform.axios({
+      url: `http://${this.device.deviceUrl}/config`,
+      method: 'post',
+      data: this.Data,
+    },
+    );
+    if (this.On) {
+      this.Request = 'On';
+    } else {
+      this.Request = 'Off';
+    }
+    this.platform.log.info('Sending request %s for %s', this.Request, this.accessory.displayName);
+    this.platform.log.debug('%s %s Changes pushed -', this.device.model, this.accessory.displayName, JSON.stringify(push.data));
+  }
+
+  updateHomeKitCharacteristics() {
+    switch (this.device.model) {
+      case 'MSL-100':
+      case 'MSL-120':
+      case 'MSL-320':
+      case 'MSL-420':
+        if (this.On !== undefined) {
+          this.service.updateCharacteristic(this.platform.Characteristic.On, this.On);
+        }
+        if (this.Brightness !== undefined) {
+          this.service.updateCharacteristic(this.platform.Characteristic.Brightness, this.Brightness);
+        }
+        if (this.Saturation !== undefined) {
+          this.service.updateCharacteristic(this.platform.Characteristic.Saturation, this.Saturation);
+        }
+        if (this.ColorTemperature !== undefined) {
+          this.service.updateCharacteristic(this.platform.Characteristic.ColorTemperature, this.ColorTemperature);
+        }
+        if (this.Hue !== undefined) {
+          this.service.updateCharacteristic(this.platform.Characteristic.Hue, this.Hue);
+        }
+        break;
+      case 'MSS560':
+      default:
+        if (this.On !== undefined) {
+          this.service.updateCharacteristic(this.platform.Characteristic.On, this.On);
+        }
+        if (this.Brightness !== undefined) {
+          this.service.updateCharacteristic(this.platform.Characteristic.Brightness, this.Brightness);
+        }
+    }
+  }
+
+  public apiError(e: any) {
+    switch (this.device.model) {
+      case 'MSL-100':
+      case 'MSL-120':
+      case 'MSL-320':
+      case 'MSL-420':
+        this.service.updateCharacteristic(this.platform.Characteristic.On, e);
+        this.service.updateCharacteristic(this.platform.Characteristic.Brightness, e);
+        this.service.updateCharacteristic(this.platform.Characteristic.Saturation, e);
+        this.service.updateCharacteristic(this.platform.Characteristic.ColorTemperature, e);
+        this.service.updateCharacteristic(this.platform.Characteristic.Hue, e);
+        break;
+      case 'MSS560':
+      default:
+        this.service.updateCharacteristic(this.platform.Characteristic.On, e);
+        this.service.updateCharacteristic(this.platform.Characteristic.Brightness, e);
+    }
+  }
+
+  /**
+   * Handle requests to set the value of the "On" characteristic
+   */
+  OnSet(value: CharacteristicValue) {
+    this.platform.log.debug('%s %s - Set On: %s', this.device.model, this.accessory.displayName, value);
+
+    this.On = value;
+    this.doUpdate.next();
```

Comment:
Your doUpdate() get's called here which causes an update of the brightness to. That's why we are getting 4 log messages even though we are only updating the "On"-state. Also, because your are using "this.Request = ..." and calling the update methods async, you are overriding the value. That's why we are getting 4x the same log message.

Reply:
Regarding the async stuff and overriding the value:

- That was not correct, the log messages were just wrong (copy pasted?).